### PR TITLE
[DX-1305] [TT-9514] Include tyk-operator chart as optional components in the umbrella chart

### PR DESCRIPTION
**Pull request by buraksekili**
### For internal users - Please add a Jira DX PR ticket to the subject!
<be>

https://tyktech.atlassian.net/browse/TT-9514
https://tyktech.atlassian.net/browse/DX-1305

---

### Preview Link

https://deploy-preview-4548--tyk-docs.netlify.app/docs/nightly

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labelled** the PR

___

## **Type**
enhancement

___

## **Description**
- Added Tyk Operator as an optional component in the Tyk Control Plane, OSS, and Stack charts.
- Introduced new configuration sections in each chart's documentation for enabling and setting up Tyk Operator.
- Updated the Tyk Operator installation guide to include options for installing as a sub-chart and corrected a typo.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>tyk-control-plane-chart.md</strong><dd><code>Include Tyk Operator as Optional Component in Control Plane Chart</code></dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md
<li>Updated the components table to include Tyk Operator as an optional <br>component.<br> <li> Added a new section detailing the configurations for enabling and <br>setting up Tyk Operator.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4548/files#diff-1bb3ae7ce6b5aec344682558fe28a0e91ef5504473fbed884ae4651089784cf7">+19/-7</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-oss-chart.md</strong><dd><code>Include Tyk Operator as Optional Component in OSS Chart</code>&nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-oss-chart.md
<li>Updated the components table to include Tyk Operator as an optional <br>component.<br> <li> Added a new section detailing the configurations for enabling and <br>setting up Tyk Operator.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4548/files#diff-53fd570dd4fbaa9b39eca159ac973030b7f820da118b68df42df490432741dd5">+15/-4</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-stack-chart.md</strong><dd><code>Include Tyk Operator as Optional Component in Stack Chart</code></dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-stack-chart.md
<li>Updated the components table to include Tyk Operator as an optional <br>component.<br> <li> Added a new section detailing the configurations for enabling and <br>setting up Tyk Operator.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4548/files#diff-f6e45bbf530fa763d587dec798584c8eb6294229ef353d3212ab01bf6b0a1d7c">+19/-6</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>installing-tyk-operator.md</strong><dd><code>Update Installation Options and Fix Typo in Tyk Operator Docs</code></dd></summary>
<hr>

tyk-docs/content/tyk-stack/tyk-operator/installing-tyk-operator.md
<li>Corrected a typo in the compatibility note.<br> <li> Expanded the installation section to include options for installing <br>Tyk Operator as a sub-chart.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4548/files#diff-0b4360968306ecca2c98ec75cb33de17643269cf4ce999c6fd93b789df611d3a">+13/-2</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Review by olamilekan000 — APPROVED** [review]

File: tyk-docs/content/product-stack/tyk-charts/tyk-oss-chart.md [hunk 1]
```diff
@@ -471,3 +472,13 @@ Uptime Pump can be configured by setting `tyk-pump.pump.uptimePumpBackend` in va
 
 #### Other Pumps
 To setup other backends for pump, refer to this [document](https://github.com/TykTechnologies/tyk-pump/blob/master/README.md#pumps--back-ends-supported) and add the required environment variables in `tyk-pump.pump.extraEnvs`
+
+### Tyk Operator Configurations
+
+In order to enable installing Tyk Operator along-side Tyk OSS installation, please set `global.components.operator`
+to `true`.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-charts/tyk-oss-chart.md]: move onto same line in line 478

File: tyk-docs/content/product-stack/tyk-charts/tyk-oss-chart.md [hunk 2]
```diff
@@ -471,3 +472,13 @@ Uptime Pump can be configured by setting `tyk-pump.pump.uptimePumpBackend` in va
...
+
+All other configurations related to Tyk Operator are available under `tyk-operator` section of `values.yaml` file.
+
+> Tyk Operator needs a cert-manager to be installed. Ensure that cert-manager is installed as described in the
+> official documentation: [Installing Tyk Operator]({{<ref "tyk-stack/tyk-operator/installing-tyk-operator">}}).
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-charts/tyk-oss-chart.md]: remove line break, include as line 483

File: tyk-docs/content/product-stack/tyk-charts/tyk-stack-chart.md [hunk 3]
```diff
@@ -857,3 +860,13 @@ tyk-dev-portal:
   - name: PORTAL_LOG_LEVEL
     value: debug
 ```
+
+### Tyk Operator Configurations
+
+In order to enable installing Tyk Operator along-side Tyk Stack installation, please set `global.components.operator`
+to `true`.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-charts/tyk-stack-chart.md]: same comments as above

File: tyk-docs/content/tyk-stack/tyk-operator/installing-tyk-operator.md [hunk 4]
```diff
@@ -176,14 +176,25 @@ $ kubectl create secret -n tyk-operator-system generic tyk-operator-conf \
...
+You can install CRDs and Tyk Operator using the stand-alone Helm Chart by running the following command:
 
 ```console
 $ helm repo add tyk-helm https://helm.tyk.io/public/helm/charts/
 $ helm repo update
 $ helm install tyk-operator tyk-helm/tyk-operator -n tyk-operator-system
 ```
 
+This process will deploy Tyk Operator and its required Custom Resource Definitions (CRDs) into your Kubernetes cluster in
+`tyk-operator-system` namespace.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-operator/installing-tyk-operator.md]: remove line break so sentence part of line 196

**Review by dcs3spp — CHANGES_REQUESTED** [review]
Hi @buraksekili Thanks for submitting PR, I have reviewed and made suggestions to fix the line breaks and once peer approved then will merge and release.

**Review by dcs3spp — COMMENTED** [review]
@buraksekili Could you add a label to the PR to indicate the version of docs (5.3, 5.2 etc) that you wish to release to? Does this relate to Tyk Charts 1.4.0?

@caroltyk would you be able to peer review?

**Review by carolwkc — APPROVED** [review]
Looks good to me!

**Comment by buraksekili** [comment]
thanks for suggestions @dcs3spp! I applied code suggestions accordingly. Could you please have a look once again?

This PR is for tyk-charts 1.5.0 release, not for 1.4.0. So, we need to wait for 1.5.0

**Review by dcs3spp — APPROVED** [review]
Thanks @buraksekili Will release PR when Tyk Charts 1.5.0 released

**Pull request merged**